quickwit-search/src/client_pool/search_client_pool.rs

```diff
@@ -129,7 +130,11 @@ impl ClientPool for SearchClientPool {
             // TODO optimize the case where there are few jobs and many clients.
             let clients = self.clients.read().await;
 
-            for (grpc_addr, client) in clients.iter() {
+            for (grpc_addr, client) in clients.iter().filter(|(grpc_addr, _)| {
+                address_whitelist
+                    .as_ref()
+                    .map_or(true, |whilelist| whilelist.contains(grpc_addr))
```

Comment:
Why the hell does `map_or` exist      (   O_O  " )

Reply:
on Option map inner value, or on none